### ISSUE TITLE
Add `MoveService` account.migrations.last to Tootctl to help with abandoned followers on remote instances

### ISSUE THREAD

**Issue by krisnova** [mastodon/mastodon#22281]
### Pitch

When migrating from one instance to another it is possible for followers to be abandoned or "left behind" on an original instance. I believe we can add a feature to `tootctl` that will allow operators of the instance in which the followers are "stuck" to manually renegotiate with the new instance. 

I believe this is because of the public key/private key negotiation that needs to happen between the 3 instances in consideration.

 - **Origin** Original host instance, the original home instance of the migrating user.
 - **Desired** New desired host instance, the intended new home for the migrating user.
 - **Remote** Remote "follower" instance, the 3rd instance that had originally followed **Origin** and after the migration should be following **Remote**.

Based off of https://github.com/mastodon/mastodon/issues/21442#issuecomment-1327954053 I believe we can add a feature to `tootctl` to call the following Ruby code:

```ruby
MoveService.new.call(account.migrations.last)
```

Which should make it easier for remote instance operators to run and fix the abandoned followers problem. Right now, unless something causes the missed followers to renegotiate, I believe they will be "stuck" forever.

### Motivation

The motivation is to give operators **something** they can do, in order to address the (what I believe is a very large) problem of followers being orphaned/abandoned during a migration when negotiation fails.

**Comment by krisnova** [comment]
If the feature is not accepted to `tootctl` directly, can we please provide documentation/best practices for instance admins to run this directly? Does Mastodon have a preferred way of executing Ruby code like this against a running system? If so -- can we please identify where this documentation should go so we can document it.

**Comment by ineffyble** [comment]
@ClearlyClaire fyi

**Comment by ClearlyClaire** [comment]
I think having such a tool would be a good idea, but I am not completely sure which failure case would cause the particular behavior being described in https://github.com/hachyderm/community/issues/74.

I will investigate, but in the meantime, here is some context on what happens when someone migrates from a server to another one is that:
1. the old server sends a `Move` to the existing followers
2. each of the followers' server will process the `Move` by fetching the new account's information, then when it has validated that the new account has an alias for the original account, queue a worker to perform the move on its side
3. the worker will queue the follow-unfollow flow for each local follower, and do a bunch of other stuff
4. each worker queued in 3. will queue a follow request and unfollow the original account

There are multiple known issues with this:
- 2. means that all servers on which the original account has followers will request the new account as approximately the same time, which may become overwhelmed; this will be retried, but every server will retry at approximately the same time as well, which will result in waves instead of being spread out properly
- 4. have the same issue than 2. except much worse because it's not one request per server anymore, but one request per follower; there is no good way around that, but #21956 aims to spread retries much more evenly
- 4. also has another issue where one possible failure state is that the new server never receives the follow request, and the follower still unfollows the old account; #21957 aims to address that

**Comment by ClearlyClaire** [comment]
Actually, there is an issue I overlooked with this proposal: currently, only the originating server stores an `AccountMigration`, so the proposed `tootctl` command could only be run on the originating server, not on a server where there are unmigrated followers as it is being proposed.

Receiving servers do not have anything in their database to have a distinction between a redirection (just setting `movedTo`) and a migration (sending a `Move` and forcibly moving followers).

**Comment by krisnova** [comment]
Would the code in the suggestion above still be applicable to be ran on the **Origin** server? 

I am wondering if that would offer the guarantees we need to ensure that the followers are migrated.

In general I think the renegotiation between the 3 instances needs work so we can see and control the process more granularly. I am not sure that the account migrations would be thorough enough by calling the code I suggested. Maybe we need to add a date range or something to get complete data out of the database.

**Comment by ClearlyClaire** [comment]
> Would the code in the suggestion above still be applicable to be ran on the **Origin** server?

Yes

> I am wondering if that would offer the guarantees we need to ensure that the followers are migrated.

Not until #21957 or similar gets merged, because that failure case would have the follower stop following the original account, without following the new one.

> I am not sure that the account migrations would be thorough enough by calling the code I suggested.

It would make the server replay the last migration of the targeted account, which includes sending the `Move` to all people currently following them. That being said, now that I'm thinking about it, there should probably be a safeguard to check that the account is still redirecting to the remote one, to prevent accidental uses where the account has cancelled their former migration.
